crates/db_schema/src/impls/comment.rs

```diff
@@ -67,13 +78,21 @@ impl Comment {
       .run(|conn| {
         Box::pin(async move {
           // Insert, to get the id
-          let inserted_comment = insert_into(comment::table)
-            .values(comment_form)
-            .on_conflict(comment::ap_id)
-            .do_update()
-            .set(comment_form)
-            .get_result::<Self>(conn)
-            .await?;
+          let inserted_comment = if let Some(timestamp) = timestamp {
+            insert_into(comment::table)
+              .values(comment_form)
+              .on_conflict(comment::ap_id)
+              .filter_target(coalesce(comment::updated, comment::published).lt(timestamp))
+              .do_update()
+              .set(comment_form)
+              .get_result::<Self>(conn)
```

Comment:
not sure but I think this will result in an error if the value is not updated? as in, the query will return zero rows which will cause get_result to return a row not found error unless you add `.optional()`

Reply:
Correct

Reply:
That should be fine, if we are receiving an old comment while already storing a newer one, it throws an error and gets rejected. We dont need to do anything in that case.